### DISCUSSION TITLE
2023/06/19/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%9514-%E6%9C%80%E9%95%BF%E5%85%AC%E5%85%B1%E5%89%8D%E7%BC%80/

### DISCUSSION THREAD

**Discussion by giscus** [Zhuanjier/blog#19] [Announcements]
# 2023/06/19/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%9514-%E6%9C%80%E9%95%BF%E5%85%AC%E5%85%B1%E5%89%8D%E7%BC%80/

¶找到二叉树中最近的右侧节点  原题传送门  编写一个函数来查找字符串数组中的最长公共前缀。 如果不存在公共前缀，返回空字符串""。 ¶示例 ¶示例1 输入：strs = [“flower”,“flow”,“flight”] 输出：“fl” ¶示例2 输入：strs = [“dog”,“racecar”,“car”] 输出：“” 解释：输入不存在公共前缀。 提示：  1 &lt

https://blog.zhuanjie.ltd/2023/06/19/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%9514-%E6%9C%80%E9%95%BF%E5%85%AC%E5%85%B1%E5%89%8D%E7%BC%80/

**Comment by RandallAnjie** [comment]
`sort`函数是C++的标准模板库(STL)中的一种算法，主要用于对序列进行排序。它需要两个迭代器参数，分别指向要排序的序列的开始和结束位置。在排序完成后，该序列将按升序排列。

在代码中，`sort(str.begin(),str.end());`这一行将`str`这个向量中的所有元素（也就是字符串）进行排序。`str.begin()`返回一个迭代器，指向向量的第一个元素，而`str.end()`返回一个迭代器，指向向量的最后一个元素后的位置。因此，`sort`函数将对从第一个元素到最后一个元素的全部元素进行排序。

`sort`函数在这里之所以能用，是因为C++标准库为字符串定义了小于操作符`<`，用于比较两个字符串的字典顺序。当你在`sort`函数中使用字符串向量时，它将使用这个小于操作符对字符串进行比较和排序。

此外，排序是为了让查找公共前缀的操作更加高效。经过排序后，最小的字符串和最大的字符串肯定包含了所有字符串的公共前缀（如果存在的话）。这样，我们就只需要比较这两个字符串，而不必遍历整个字符串列表，从而降低了计算复杂度。
